src/wallet/wallet2.h

```diff
@@ -1386,6 +1386,8 @@ namespace tools
     uint64_t credits() const { return m_rpc_payment_state.credits; }
     void credit_report(uint64_t &expected_spent, uint64_t &discrepancy) const { expected_spent = m_rpc_payment_state.expected_spent; discrepancy = m_rpc_payment_state.discrepancy; }
 
+    static const std::string &get_default_daemon_address() { CRITICAL_REGION_LOCAL(default_daemon_address_lock); return default_daemon_address; }
```

Comment:
This should be return by copy otherwise the lock does nothing. The `const std::string&` above is still valid (obscure C++ rule).